placeholder.go

```diff
@@ -12,40 +12,63 @@ import (
 // placeholder with a (possibly different) SQL placeholder.
 type PlaceholderFormat interface {
 	ReplacePlaceholders(sql string) (string, error)
+	PlaceholderHolder
+}
+
+type PlaceholderHolder interface {
+	GetPlaceholder() string
 }
 
 var (
 	// Question is a PlaceholderFormat instance that leaves placeholders as
 	// question marks.
-	Question = questionFormat{}
+	Question = questionFormat{"?"}
 
 	// Dollar is a PlaceholderFormat instance that replaces placeholders with
 	// dollar-prefixed positional placeholders (e.g. $1, $2, $3).
-	Dollar = dollarFormat{}
+	Dollar = dollarFormat{"$"}
 
 	// Colon is a PlaceholderFormat instance that replaces placeholders with
 	// colon-prefixed positional placeholders (e.g. :1, :2, :3).
-	Colon = colonFormat{}
+	Colon = colonFormat{":"}
 )
 
-type questionFormat struct{}
+type questionFormat struct {
+	value string
```

Comment:
I don't think these new fields are necessary. You can instead stick the constants into the new methods:
```go
func (questionFormat) debugPlaceholder() string {
    return "?"
}
```

Reply:
just curious, does your preference to return the constants stem from a functional vs. OO perspective?

Reply:
It stems from a "less code is better" perspective. :smile: